packages/data-context/src/actions/ProjectActions.ts

```diff
@@ -531,4 +539,52 @@ export class ProjectActions {
     // @ts-ignore - because of the conditional above, we know that devServer isn't a function
     return WIZARD_FRAMEWORKS.find((framework) => framework.configFramework === config?.component?.devServer.framework)
   }
+
+  async getDefaultSpecFileName (
```

Comment:
I moved this from `ProjectDataSource` to here and passed anything that we needed from context as props instead. This way I could reuse the logic for determining the correct file path for a spec.

Reply:
This function doesn't rely on `ctx` (or rather `this.ctx`). Refactoring this into a utils file would allow the `SpecOptions` class to simply import it and you wouldn't have to pass a function in.